api_app/script_analyzers/observable_analyzers/inquest.py

```diff
@@ -0,0 +1,62 @@
+import logging
+import requests
+
+from api_app.exceptions import AnalyzerRunException
+from api_app.script_analyzers.classes import ObservableAnalyzer
+from intel_owl import secrets
+
+
+logger = logging.getLogger(__name__)
+
+
+class InQuest(ObservableAnalyzer):
+    base_url: str = "https://labs.inquest.net"
+
+    def set_config(self, additional_config_params):
+        self.api_key_name = additional_config_params.get(
+            "api_key_name", "INQUEST_API_KEY"
+        )
+        self.__api_key = secrets.get_secret(self.api_key_name)
+        self.analysis_type = additional_config_params.get(
+            "inquest_analysis", "dfi_search"
+        )
+        self.hash_type = additional_config_params.get("hash_type", "md5")
```

Comment:
`hash_type` should not be kept static as a config var but instead we should detect the kind of hash from the user provided `observable_name`. All supported types are given in their docs, [here](https://labs.inquest.net/docs/?shell#api-dfi-search-hash-source). A utility function which detects the kind of hash should be pretty easy to find on stackoverflow, add it under `/api_app/script_analyzers/utils.py`

Reply:
Should I add it to the utils? Since, given hash types [`md5`, `sha1`, `sha256`, `sha512`] are different lengths, I can write a small helper function only for this analyzer.
```python
def __identify_hash(self, hash):
    hash_lengths = {32: "md5", 40: "sha1", 64: "sha256", 128: "sha512"}
    hash_type = hash_lengths.get(len(hash), None)
    if not hash_type:
        raise AnalyzerConfigurationException(
            f"Given Hash: '{hash}' is not suported."
            "Supported hash types are: 'md5', 'sha1', 'sha256', 'sha512'."
        )
    return hash_type
```

Incase, we need more precise hash identifier, I'll use [this](https://github.com/blackploit/hash-identifier/blob/master/hash-id.py) and move it to `api_app/script_analyzers/hash-id.py`

Reply:
for now, we can just go with the small helper function

Reply:
one more thing, can you add this `hash_type` to the `result` dictionary? like: `self.report["hash_type"] = hash_type`